packages/components/src/autocomplete/autocompleter-ui.tsx

```diff
@@ -27,11 +27,59 @@ import { VisuallyHidden } from '../visually-hidden';
 import { createPortal } from 'react-dom';
 import type { AutocompleterUIProps, KeyedOption, WPCompleter } from './types';
 
+type ListBoxProps = {
+	items: KeyedOption[];
+	onSelect: ( option: KeyedOption ) => void;
+	selectedIndex: number;
+	instanceId: number;
+	listBoxId: string | undefined;
+	className?: string;
+	Component?: React.ElementType;
+};
+
 export function getAutoCompleterUI( autocompleter: WPCompleter ) {
 	const useItems = autocompleter.useItems
 		? autocompleter.useItems
 		: getDefaultUseItems( autocompleter );
 
+	function ListBox( {
+		items,
+		onSelect,
+		selectedIndex,
+		instanceId,
+		listBoxId,
+		className,
+		Component = 'div',
+	}: ListBoxProps ) {
+		return (
+			<Component
+				id={ listBoxId }
+				role="listbox"
+				className="components-autocomplete__results"
+			>
+				{ items.map( ( option, index ) => (
+					<Button
+						key={ option.key }
+						id={ `components-autocomplete-item-${ instanceId }-${ option.key }` }
+						role="option"
+						aria-selected={ index === selectedIndex }
+						disabled={ option.isDisabled }
+						className={ clsx(
+							'components-autocomplete__result',
+							className,
+							{
+								'is-selected': index === selectedIndex,
+							}
+						) }
+						onClick={ () => onSelect( option ) }
+					>
+						{ option.label }
+					</Button>
+				) ) }
+			</Component>
+		);
+	}
```

Comment:
Do we need it to be inside of `getAutoCompleterUI`?

Reply:
It can be outside! Moved 🙂